tsdb/shard_test.go

```diff
@@ -203,9 +208,27 @@ func TestMaxSeriesLimit(t *testing.T) {
 		t.Fatal("expected error")
 	} else if exp, got := `partial write: max-series-per-database exceeded limit=1000 series=1000 keys=1 dropped=1`, err.Error(); exp != got {
 		t.Fatalf("unexpected error message:\n\texp = %s\n\tgot = %s", exp, got)
+	} else {
+		st := sh.Statistics(map[string]string{})
+		for _, stat := range st {
+			if stat.Name == "shard" {
+				checkInt64Stat(t, stat, "writePointsOk", int64(opts.Config.MaxSeriesPerDatabase))
+				checkInt64Stat(t, stat, "writePointsDropped", 1)
+				return
```

Comment:
I'm slightly worried about returning in the middle of a test case. If someone adds some code onto the end of the test case and doesn't see the return on success, they might think their new code works when it really doesn't. Would it make sense to restructure the code so there's not a return in the for loop?

Reply:
I am not bothered by this, because in tests we don't use `return` to hand back errors, we call `t.Fatal` or `t.Fatalf` so it's actually worse than you think, because we are killing ourselves in the `checkInt64Stat` method.

Reply:
I think the issue @gwossum is taking about is if someone were to call `t.Fatalf` somewhere below here - their test would pass because we returned here and never got to `t.Fatalf`.

But also, I'm not sure this test is actually helping. I misunderstood the first PR when I asked for this test - I'm also fine if we just close this current PR and pretend I never stuck my nose in :)

Reply:
@lesam is correct, I'm worried about someone adding test code after the loop and not realizing it won't run. It looks like it may not matter.